### ISSUE TITLE
[Is this possible?] Running tasks in parallel, e.g. pilot + zip upload

### ISSUE THREAD

**Issue by cbowns** [fastlane/fastlane#10241]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm looking at parallelizing a couple of steps in our main fastlane action for uploading a build to TestFlight for testing:

I want to run `pilot`, which I have set to synchronously check on build processing and set the changelog (n.b. I have just added this to the lane and not yet tested that it actually works). While that's running, I want to do some post-upload things:

- zip up the built archive
- upload that zip to S3 for safe keeping
- maybe other things like make me a sandwich

My question: is this possible? Have I reached the edge of the world by wanting this? Am I thinking about this all wrong? "What am I really trying to do?"

**Comment by KrauseFx** [comment]
Sorry, currently not possible, you could hack something together using the normal Ruby syntax using `Thread.new`, but this might have side effects, I'd advice against it

**Comment by cbowns** [comment]
that's what I figured.
